### ISSUE TITLE
Helm warning when using gha-runner-scale-set as app dependency

### ISSUE THREAD

**Issue by rfinnie-epic** [actions/actions-runner-controller#3620]
### Checks

- [X] I've already read https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/troubleshooting-actions-runner-controller-errors and I'm sure my issue is not covered in the troubleshooting guide.
- [X] I am using charts that are officially provided

### Controller Version

0.9.2

### Deployment Method

Helm

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions/actions-runner-controller/discussions)).
- [X] I've read the [Changelog](https://github.com/actions/actions-runner-controller/blob/master/docs/gha-runner-scale-set-controller/README.md#changelog) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes

### To Reproduce

`Chart.yaml`:

```yaml
---
apiVersion: 'v2'
name: 'arc-runners'
description: 'GitHub arc runners'
type: 'application'
version: '0.1.0'

dependencies:
  - name: 'gha-runner-scale-set'
    version: '0.9.2'
    repository: 'oci://ghcr.io/actions/actions-runner-controller-charts'
```

`values.yaml`:

```yaml
---
gha-runner-scale-set:
  githubConfigUrl: 'https://github.com/enterprises/example'
  githubConfigSecret: 'runner-pat'
  # ...
```

Run:
```shell
helm install my-label --namespace arc-runners --values ./values.yaml .
```

### Describe the bug

When the above chart is installed or upgraded while pulling in gha-runner-scale-set as a dependency, it produces the following Helm warning:

```
Release "my-label" does not exist. Installing it now.
coalesce.go:286: warning: cannot overwrite table with non table for arc-runners.gha-runner-scale-set.githubConfigSecret (map[github_token:])
NAME: my-label
LAST DEPLOYED: Sun Jun 23 15:30:22 2024
NAMESPACE: arc-runners
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

This warning appears to be harmless as it does successfully use the `runner-pat` k8s secret, but I'm not sure if it's a error on the upstream chart, or with Helm itself.

### Describe the expected behavior

No warning

### Additional Context

```yaml
See above
```

### Controller Logs

```shell
N/A
```

### Runner Pod Logs

```shell
N/A
```

**Comment by keyluck** [comment]
Are there any workarounds for this issue?

**Comment by arseny-zinchenko** [comment]
Got the same weird warning.
But that issue was opened Jun 24, and still no comments from the GitHub team.

**Comment by nikola-jokic** [comment]
Hey,

I think your configuration is incorrect. 

The [githubConfigSecret](https://github.com/actions/actions-runner-controller/blob/2f5c981d4699398b28026f29412feded27ffe743/charts/gha-runner-scale-set/values.yaml#L12) is an object. You specified a string (`githubConfigSecret: 'runner-pat'`).

**Comment by nikola-jokic** [comment]
Seems like a duplicate of #3819

**Comment by nikola-jokic** [comment]
Hey everyone, can you please help me understand how to reproduce this issue? Please write down the steps to reproduce it.

**Comment by dheitman-prom** [comment]
Just ran into this myself.

tl;dr: if you use this chart as a dependency of your own chart, and attempt to use variation C of providing a value for githubConfigSecret, helm throws a warning because you're attempting to overwrite a map with a string. The verbiage of the warning is confusing, as it says it "cannot" perform the overwrite, but does anyway.  This also occurs with helm template.

The process to reproduce is exactly what the OP says.  You create a chart that includes the gha-runner-scale-set chart as a dependency (Chart.yaml from the first post in this issue).  Then, you attempt to set values for the gha-runner-scale-set dependency chart (seen in the values.yaml from the first post), specifically using Variation C, a pre-defined k8s secret for githubConfigSecret.  Helm template will throw a warning about overwriting a table with a non-table.

The reason for this is this line: https://github.com/actions/actions-runner-controller/blob/a1a8dc5606d7d4e5532994f7c4a245354384d8b3/charts/gha-runner-scale-set/values.yaml#L15

Because "github_token" is uncommented in the default values.yaml for the chart, now "githubConfigSecret" is a map, and attempting to override that with a string (for variation C, using a pre-defined k8s secret) results in this error.  Given that the current default value for github_token (an empty string) is of no use, it would make sense to comment out github_token entirely, and leave it as a mandatory exercise for the user to uncomment and populate an appropriate value or set of values (depending on the variation used).

Alternately, the chart could be changed so that even Variation C uses a map.  A and B already do (githubConfigSecret.github_token for variation A, and for B it's app_id, app_installation_id, and app_private_key), C could also use a map, something like githubConfigSecret.secret_name, for example, then all three variations use a map. Either approach is fine, but I'd like to avoid maintaining a local fork of this chart.

**Comment by nikola-jokic** [comment]
Closing this one since it is basically similar to https://github.com/actions/actions-runner-controller/issues/3819, and the mentioned issue has more interaction on it.
